redis/src/aio/connection.rs

```diff
@@ -380,8 +386,22 @@ pub(crate) async fn connect_simple<T: RedisRuntime>(
 ) -> RedisResult<T> {
     Ok(match connection_info.addr {
         ConnectionAddr::Tcp(ref host, port) => {
-            let socket_addr = get_socket_addrs(host, port).await?;
-            <T>::connect_tcp(socket_addr).await?
+            let socket_addrs = get_socket_addrs(host, port).await?;
+            stream::iter(socket_addrs)
+                .map(|socket_addr| async move { <T>::connect_tcp(socket_addr).await })
+                .fold(
```

Comment:
Truly minor, feel free to ignore - 
can you either remove the `map` and call `connect_tcp` in the `fold` operation?
Alternatively, can the fold operation be moved to function, to reduce repetition?